packages/firestore/src/local/local_store_impl.ts

```diff
@@ -1526,3 +1526,38 @@ export async function localStoreConfigureFieldIndexes(
         .next(() => PersistencePromise.waitFor(promises))
   );
 }
+
+export function localStoreSetIndexAutoCreationEnabled(
+  localStore: LocalStore,
+  isEnabled: boolean
+): void {
+  const localStoreImpl = debugCast(localStore, LocalStoreImpl);
+  localStoreImpl.queryEngine.indexAutoCreationEnabled = isEnabled;
+}
+
+/**
+ * Test-only hooks into the SDK for use exclusively by tests.
+ */
+export class TestingHooks {
```

Comment:
Do we still need it here?

Reply:
This `TestingHooks` class is used by the unit tests (as opposed to the integration tests). It doesn't go through the more complicated testing hooks mechanism to make it available to integration tests.

An alternative to this `TestingHooks` class is to instead add a top-level function named something like `localStoreSetIndexAutoCreationSettings()`. But then it's not clear that the function exists solely for use by tests. Do you think that would be better?

Reply:
I'm fine with keeping it then.